### DISCUSSION TITLE
How to set batch_size in other communication method e.q LoRa

### DISCUSSION THREAD

**Discussion by likehbbfoe** [NVlabs/sionna#466] [Q&A] [answered]
if I have a signal，the shape of the signal is (N,1) with type as tf.complex64 and it include N/1024 symbols. How do I input this data into a time domain model to simulate signal variations. Such How reshape the data into (batch_size, num_rx, num_nx, num_time_samples).what is the meaning of the batch_size?

**Comment by SebastianCa** [comment]
Hi @likehbbfoe,

Please take a look at the [Sionna tutorial notebook](https://nvlabs.github.io/sionna/examples/Sionna_tutorial_part1.html) to learn more about the batch size and shapes used in Sionna. For additional details about time domain simulations, refer to the [MIMO OFDM notebook](https://nvlabs.github.io/sionna/examples/MIMO_OFDM_Transmissions_over_CDL.html#Uplink-Transmission-in-the-Time-Domain).

I hope this helps!

**Reply by likehbbfoe** [reply]
ok,thank you for your answer. I can set the right size with my LoRa signal and simulate the transmission process. And I also have a question is that in our real estimation, the SNR is -10dB. Under the SNR, we also get the BER in real is 0, but in sionna we use the same physical algorithm to simulate.
noise is set by this:
`ebno_db = 10. # SNR in dB`
`no = ebnodb2no(ebno_db,
               1,
               1)`
and use time domain simulation:
`channel = ApplyTimeChannel(num_time_samples = 37120 ,l_tot=14,add_awgn=True)`
`y = channel((s_np, h_m,no))`
the shape of my sig is (37120,1)
the output signal is full of the noise that we can not decode any information in sionna. Only when I set the `ebno_db ` to 100 and we can get the right information.
I think the time domain simulation is just simulating of electromagnetic wave transmission process. So what is the reason for such a big difference between simulation and real world.

**Reply by likehbbfoe** [reply]
one more thing is if I just use follow code: 
`awgn_channel = AWGN() `
`ebno_db = 10. # SNR in dB` 
`no = ebnodb2no(ebno_db,
               1,
               1) `
`y = awgn_channel((sig, no))`
I can get the correct infomation. I do not know what lead to the difference.

**Comment by SebastianCa** [comment]
Hi @likehbbfoe,
It's difficult to provide detailed advice without knowing the specifics of your setup. 

Did you ensure that `h_m` is normalized to unit power? If not, the SNR definition might be incorrect, which could explain why you see shifted curves.

Keep in mind that the first case is a more sophisticated channel model, typically requiring channel estimation and equalization at the receiver.

**Reply by likehbbfoe** [reply]
Thank you for your help, your advice is very useful and helped me to solve this problem. One more thing is that in the document of Sionna, the funtion `ebnodb2no` seem to be designed for OFDM. If I use this function for LoRa simulation， the noise impacts are same or different, compare to the common noise impacts. 
My setting is `no = ebnodb2no(ebno_db, 1, 1)  # ebno_db(SNR) is 5dB`
Any advice is useful for my work, thank you!

**Reply by SebastianCa** [reply, marked as answer]
Hi @likehbbfoe,
I don't know the exact details of your LoRa simulations However, the `ebnodb2no` function is only required as utility to calculate the rate-adjusted SNR (i.e., account for the rate-loss due to piloting etc.). In your above code snippet you effectively evaluate the non-rate adjusted SNR (i.e, $E_s/N_0$). Alternatively, you need to provide the code rate and modulation order of your system.
